mbuild/compound.py

```diff
@@ -956,14 +956,24 @@ def bonds(self):
 
     @property
     def n_bonds(self):
```

Comment:
I think we need to be careful here, since this method now is giving two types of information. As in, if the compound object is a container, we are returning the number of bonds that it contains (but not bond that connects it to a different compound); if the compound is a particle, it's returning the bonds that that particle is a member of. I am worry that this may cause some confusion among user, so I think we could either explain this really well in the documents and throw a warning whenever the Compound object this method called upon is a particle (like `This is a Particle so this method is returning all bonds that this particle is involved`), or we can also split this into two different methods. At the moment I still not sure which approach would be better, so may be a good discussion point for all the dev @mosdef-hub/mosdef-contributors

Reply:
Yeah, the amount of warning may be a bit too much. From the top of my head, I don't remember if there is a method that give out different type of info based on the hierarchy of the Compound. But there are methods that are only valid for certain level of Compound (e.g. `Compound.bond_graph()` only exists at the top level, and is `None` everywhere else), so I think we can have something similar, by having new method to list out bonds that a particle is involved in (`parmed.Atom` has something similar called `bond_partners`).

Reply:
I agree it's probably best to add separate method for a particle, and returning `n_bonds` back to its original behavior.

Reply:
Has this behavior been implemented yet? Overall this is looking pretty nice, I'd say we should change this doc string now to say:
`"""Return the total number of bonds contained in the Compound."""`.
and the number of direct_bonds would be described as:
`"""Return the number of bonds that contain the Particle."""`.

Reply:
Ok, I created a `direct_bonds` function that can be called on a particle rather than the `n_bonds` function.  

I guess a couple remaining questions are:

1. Do we want `direct_bonds` to return the bonding compounds, or just the number of bonding particles? In terms of the original issue, I think just the number of bonding particles is sufficient, but it could also be useful to return a list of the compounds a particle is bonded to.

2. `direct_bonds` throws an exception if it is called on a Compound that contains children compounds. Do we want `n_bonds` to now throw an exception when it's called on a compound that does not contain children. Right now it just returns zero. Maybe a warning instead of an exception?

Reply:
1. I think that would be a good feature to have. May be we can mirror what we have currently have for `Compound.bonds()` and `Compound.n_bonds`, so something like `Compound.direct_bonds()` will return a list of bonds and `Compound.n_direct_bonds` will return the number of direct bond. 
2. I think throwing an error may be a good tactics since this method has confused several users.